### ISSUE TITLE
Terragrunt run-all requires a "terragrunt.hcl" in current folder

### ISSUE THREAD

**Issue by Fran-Rg** [gruntwork-io/terragrunt#1540]
From https://terragrunt.gruntwork.io/docs/reference/cli-options/#run-all:
```
Example:
terragrunt run-all apply

This will recursively search the current working directory for any folders that contains Terragrunt modules and run apply in each one, concurrently, while respecting ordering defined via dependency and dependencies blocks.
```

I'm trying to run it from this tree (at demo/nonprod/):
```
.
├── bucket_role
│   └── terragrunt.hcl
├── env.hcl
├── iam_policy
│   └── terragrunt.hcl
└── iam_test_policy
    └── terragrunt.hcl
```
First bug:
When I run:
```
terragrunt run-all destroy
```
Nothing displays. The next time I run any terragrunt command:
```
terragrunt ouput
WARNING: Are you sure you want to run `terragrunt destroy` in each folder of the stack described above? There is no undo! (y/n) time=2021-02-10T16:23:48Z level=error msg=Error reading file at path demo/nonprod/terragrunt.hcl: open demo/nonprod/terragrunt.hcl: no such file or directory
time=2021-02-10T16:23:48Z level=error msg=Unable to determine underlying exit code, so Terragrunt will exit with error code 1
```
First Bug: Similarly to the "warn" issue and the new logging, nothing displays if it errors 1
Second bug: Why is it looking for a terragrunt.hcl in my current folder where they are all in sub-folders.

I'm on:
```
> terragrunt --version
terragrunt version v0.28.3
```

**Comment by brikis98** [comment]
First bug may be a dupe of the logging issues being worked on in https://github.com/gruntwork-io/terragrunt/pull/1541.

As for being able to run `run-all` without a `terragrunt.hcl` in the current folder, that does feel like something we should add support for. Is anyone up for a PR to make this change?

**Comment by giom-l** [comment]
Hi, 
I just hit this issue too today with older version of terrragrunt (`v0.23.8`).
I was expected this to behave like `terragrunt apply-all`, which works even if there is no `terragrunt.hcl` in the root folder.

**Comment by jon-flowers** [comment]
I ran into this problem as well, on 0.31.11 (I am currently in a multi-week process of upgrading to new versions of Terraform and Terragrunt).  We had not seen this before because we were using terragrunt apply-all like @giom-l , but I received the deprecation warnings saying I needed to change to `terragrunt run-all apply`.  Like others here have stated, this caused a new issue for us, because not every level of our directory has terragrunt.hcl files.

I assume I can just add blank ones and it will resolve the issue, but definitely was an unexpected issue since the old command this supposedly "replaced" didn't have this problem

**Comment by Wheway-AS** [comment]
This issue persists in Terragrunt 0.68.9.
Per the comment from @jon-flowers, I was able to work around it by creating a blank `terragrunt.hcl` in the run directory, after that it didn't give the error and pulled in the `terragrunt.hcl` files from each of the sub-folders as expected.

**Comment by yhakbar** [comment]
Hey folks, it seems like this is a pretty old issue, so I'd like for us to get resolution on it, if we can.

It is no longer best practice to name the file at the root of Terragrunt projects "terragrunt.hcl", as it complicates a lot of behavior, and it's a lot more manageable to simply name it something else (e.g. `root.hcl`):
https://terragrunt.gruntwork.io/docs/migrate/migrating-from-root-terragrunt-hcl/

It is expected behavior that users must have a `terragrunt.hcl` file in each directory where infrastructure is going to be provisioned (we call this a `unit` of infrastructure), and we have no intentions of changing that. Is that in line with everyone's expectations?

What else in this thread am I not addressing? Before answering that, please take a second and skim through the Terragrunt [terminology](https://terragrunt.gruntwork.io/docs/getting-started/terminology/) page so that we can discuss this issue with shared terminology. We've changed quite a bit of how Terragrunt works since 2021.

**Comment by Wheway-AS** [comment]
> Hey folks, it seems like this is a pretty old issue, so I'd like for us to get resolution on it, if we can.
> 
> It is no longer best practice to name the file at the root of Terragrunt projects "terragrunt.hcl", as it complicates a lot of behavior, and it's a lot more manageable to simply name it something else (e.g. `root.hcl`): https://terragrunt.gruntwork.io/docs/migrate/migrating-from-root-terragrunt-hcl/
> 
> It is expected behavior that users must have a `terragrunt.hcl` file in each directory where infrastructure is going to be provisioned (we call this a `unit` of infrastructure), and we have no intentions of changing that. Is that in line with everyone's expectations?
> 
> What else in this thread am I not addressing? Before answering that, please take a second and skim through the Terragrunt [terminology](https://terragrunt.gruntwork.io/docs/getting-started/terminology/) page so that we can discuss this issue with shared terminology. We've changed quite a bit of how Terragrunt works since 2021.

In our specific case, we do use a `root.hcl` at the top of the tree, but further down that tree we have what we refer to as "project" folders, which basically just hold a .yaml file with project-related variables, and a collection of folders that contain what we refer to as "resources", which hold the actual `terragrunt.hcl` file. We run the `terragrunt run-all` commands from within the project folder, so we can make sure it deploys all of the necessary "resources" for it and that all dependencies are correctly identified from within the resource itself.

As a visual aid, here's a regomized version of our tree for reference:
```
root
|-- root.hcl
|-- some-folder-name/
  |-- some-region/
    |-- region.yaml
    |-- some-environment/
      |-- environment.yaml
      |-- project-01/  <-- This is where we are running `terragrunt run-all`
        |-- project.yaml
        |-- resource-01/
          |-- terragrunt.hcl
        |-- resource-02/ 
          |-- terragrunt.hcl <-- Depends on resource-01
        |-- resource-03/
          |-- terragrunt.hcl  <-- Depends on resource-02
```

To recap: We run `terragrunt run-all` from the `project-01` folder, expecting it to pull in the `terragrunt.hcl` files from the sub-folders without error. Instead, it gives us an error that there is no `terragrunt.hcl` file within `project-01` itself. To work around this, we have created an empty `terragrunt.hcl` file directly under the `project-01` folder, and everything runs without error

**Comment by yhakbar** [comment]
Thanks for that!

Can you confirm that you are experiencing this behavior on the latest version of Terragrunt? You should be able to run `terragrunt run-all` anywhere in the [stack](https://terragrunt.gruntwork.io/docs/getting-started/terminology/#stack) across all [units](https://terragrunt.gruntwork.io/docs/getting-started/terminology/#unit). 

You shouldn't need a `terragrunt.hcl` file in `project-01` to have all the units picked up.

**Comment by Wheway-AS** [comment]
> Thanks for that!
> 
> Can you confirm that you are experiencing this behavior on the latest version of Terragrunt? You should be able to run `terragrunt run-all` anywhere in the [stack](https://terragrunt.gruntwork.io/docs/getting-started/terminology/#stack) across all [units](https://terragrunt.gruntwork.io/docs/getting-started/terminology/#unit).
> 
> You shouldn't need a `terragrunt.hcl` file in `project-01` to have all the units picked up.

Very interesting! I upgraded to v0.73.7 and the problem has gone away. I was able to delete the empty `terragrunt.hcl` and both `terragrunt run-all plan` and `terragrunt run-all apply` both work without error.
Thank you for following up on this!

**Comment by yhakbar** [comment]
@Fran-Rg , can you confirm that you're still experiencing this issue? If not, I'd like to close this issue out.

**Comment by puzzloholic** [comment]
Is this related to my issue [here](https://github.com/gruntwork-io/terragrunt/issues/4449)?

**Comment by Wheway-AS** [comment]
> Is this related to my issue [here](https://github.com/gruntwork-io/terragrunt/issues/4449)?

I don't believe so, it looks like your issue has to do with the `Auto-Init` feature, while the problem in here has to do with TG expecting a `terragrunt.hcl` file in the current directory when it shouldn't be necessary.
